server/jetstream_api.go

```diff
@@ -747,90 +798,110 @@ func jsNotFoundError(err error) *ApiError {
 }
 
 // Request to create a stream.
-func (s *Server) jsStreamCreateRequest(sub *subscription, c *client, subject, reply string, msg []byte) {
-	if c == nil || c.acc == nil {
+func (s *Server) jsStreamCreateRequest(sub *subscription, c *client, subject, reply string, rmsg []byte) {
+	if c == nil || !s.JetStreamIsLeader() {
+		return
+	}
+	ci, acc, _, msg, err := s.getRequestInfo(c, rmsg)
+	if err != nil {
+		s.Warnf(badAPIRequestT, msg)
 		return
 	}
 
 	var resp = JSApiStreamCreateResponse{ApiResponse: ApiResponse{Type: JSApiStreamCreateResponseType}}
-	if !c.acc.JetStreamEnabled() {
+	if !acc.JetStreamEnabled() {
 		resp.Error = jsNotEnabledErr
-		s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(&resp))
+		s.sendAPIResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
 		return
 	}
 	var cfg StreamConfig
 	if err := json.Unmarshal(msg, &cfg); err != nil {
 		resp.Error = jsInvalidJSONErr
-		s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(&resp))
+		s.sendAPIResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
 		return
 	}
 	streamName := streamNameFromSubject(subject)
 	if streamName != cfg.Name {
-		resp.Error = &ApiError{Code: 400, Description: "stream name in subject does not match request"}
-		s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(&resp))
+		resp.Error = jsStreamMismatchErr
+		s.sendAPIResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
 		return
 	}
 
-	mset, err := c.acc.AddStream(&cfg)
+	if s.JetStreamIsClustered() {
+		s.jsClusteredStreamRequest(ci, subject, reply, rmsg, &cfg)
+		return
+	}
+
+	mset, err := acc.AddStream(&cfg)
 	if err != nil {
 		resp.Error = jsError(err)
-		s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(&resp))
+		s.sendAPIResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
 		return
 	}
 	resp.StreamInfo = &StreamInfo{Created: mset.Created(), State: mset.State(), Config: mset.Config()}
-	s.sendAPIResponse(c, subject, reply, string(msg), s.jsonResponse(resp))
+	s.sendAPIResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(resp))
 }
 
 // Request to update a stream.
-func (s *Server) jsStreamUpdateRequest(sub *subscription, c *client, subject, reply string, msg []byte) {
-	if c == nil || c.acc == nil {
+func (s *Server) jsStreamUpdateRequest(sub *subscription, c *client, subject, reply string, rmsg []byte) {
+	if c == nil {
```

Comment:
shouldnt these also be `if c == nil || !s.JetStreamIsLeader() {` - few places here of that

Reply:
I need to flesh out the rest of the API, not all implemented yet.

Reply:
Marked here - https://github.com/nats-io/nats-server/projects/2

Reply:
Part of https://github.com/nats-io/nats-server/pull/1819